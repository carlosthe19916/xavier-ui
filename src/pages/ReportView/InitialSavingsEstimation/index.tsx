import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import InitialSavingsEstimation from './InitialSavingsEstimation';
import { GlobalState } from '../../../models/state';
import * as reportActions from '../../../actions/ReportActions';

const mapStateToProps = (state: GlobalState) => {
    const {
        report,
        reportInitialSavingEstimation,
        reportInitialSavingEstimationFetchStatus
    } = state.reportState;
    return {
        report,
        reportInitialSavingEstimation,
        reportInitialSavingEstimationFetchStatus
    };
};

const mapDispatchToProps = {
    fetchReportInitialSavingEstimation: reportActions.fetchReportInitialSavingEstimation
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(InitialSavingsEstimation)
);
