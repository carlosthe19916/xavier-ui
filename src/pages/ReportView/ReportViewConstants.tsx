export const ReportViewPaths = {
    workloadMigrationSummary: 'workloadMigrationSummary',
    initialSavingsEstimation: 'initialSavingsEstimation',
    workloadInventory: 'workloadInventory'
};
