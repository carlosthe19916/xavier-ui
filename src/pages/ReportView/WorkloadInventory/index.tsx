import { connect } from 'react-redux';
import WorkloadInventory from './WorkloadInventory';
import { GlobalState } from '../../../models/state';
import * as reportActions from '../../../actions/ReportActions';

const mapStateToProps = (state: GlobalState) => {
    const {
        reportWorkloadInventory,
        reportWorkloadInventoryFetchStatus,
        reportWorkloadInventoryCSVFetchStatus,
        reportWorkloadInventoryAvailableFilters,
        reportWorkloadInventoryAvailableFiltersFetchStatus
    } = state.reportState;
    return {
        reportWorkloadInventory,
        reportWorkloadInventoryFetchStatus,
        reportWorkloadInventoryCSVFetchStatus,
        reportWorkloadInventoryAvailableFilters,
        reportWorkloadInventoryAvailableFiltersFetchStatus
    };
};

const mapDispatchToProps = {
    fetchReportWorkloadInventory: reportActions.fetchReportWorkloadInventory,
    fetchReportWorkloadInventoryCSV: reportActions.fetchReportWorkloadInventoryCSV,
    fetchReportWorkloadInventoryAvailableFilters: reportActions.fetchReportWorkloadInventoryAvailableFilters
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(WorkloadInventory);
