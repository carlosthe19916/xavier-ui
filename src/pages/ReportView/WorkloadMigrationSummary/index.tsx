import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import WorkloadMigrationSummary from './WorkloadMigrationSummary';
import { GlobalState } from '../../../models/state';
import * as reportActions from '../../../actions/ReportActions';

const mapStateToProps = (state: GlobalState)  => {
    let {
        report,
        reportMigrationSummary
    } = state.reportState;
    return {
        report,
        reportMigrationSummary
    };
};

const mapDispatchToProps = (dispatch: any) => {
    return bindActionCreators({
        fetchReportWorkloadMigrationSummary: reportActions.fetchReportWorkloadMigrationSummary
    }, dispatch);
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(WorkloadMigrationSummary)
);
