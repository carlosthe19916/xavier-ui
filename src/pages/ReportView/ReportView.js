import React from 'react';
import { Redirect, Switch, Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReportViewPage from '../../PresentationalComponents/ReportViewPage';
import asyncComponent from '../../Utilities/asyncComponent';
import { ReportViewPaths } from './ReportViewConstants';

const WorkloadMigrationSummary = asyncComponent(() =>
    import(/* webpackChunkName: "WorkloadMigrationSummary" */ './WorkloadMigrationSummary'));
const InitialSavingsEstimation = asyncComponent(() =>
    import(/* webpackChunkName: "InitialSavingsEstimation" */ './InitialSavingsEstimation'));
const WorkloadInventory = asyncComponent(() =>
    import(/* webpackChunkName: "WorkloadInventory" */ './WorkloadInventory'));

class ReportView extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            reportId: props.match.params.reportId
        };
    }

    componentDidMount() {
        const { reportId } = this.state;
        this.props.fetchReport(reportId);
    }

    render() {
        const { report, reportFetchStatus } = this.props;
        return (
            <ReportViewPage
                report={ report }
                reportFetchStatus={ reportFetchStatus }
            >
                <Switch>
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.workloadMigrationSummary}` }
                        component={ WorkloadMigrationSummary }
                    />
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.initialSavingsEstimation}` }
                        component={ InitialSavingsEstimation }
                    />
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.workloadInventory}` }
                        component={ WorkloadInventory }
                    />

                    <Redirect
                        from={ `${this.props.match.url}` }
                        to={ `${this.props.match.url}/${ReportViewPaths.initialSavingsEstimation}` }
                    />
                </Switch>
            </ReportViewPage>
        );
    }
}

ReportView.propTypes = {
    match: PropTypes.object,
    report: PropTypes.object,
    reportFetchStatus: PropTypes.object,
    fetchReport: PropTypes.func
};

export default ReportView;
