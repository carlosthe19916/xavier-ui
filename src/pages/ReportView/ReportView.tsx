import React from 'react';
import { Redirect, Switch, Route } from 'react-router-dom';
import { RouterGlobalProps } from '../../models/router';
import { Report } from '../../models';
import ReportViewPage from '../../PresentationalComponents/ReportViewPage';
import asyncComponent from '../../Utilities/asyncComponent';
import { ReportViewPaths } from './ReportViewConstants';
import { FetchStatus } from '../../models/state';
import { AxiosError } from 'axios';

interface StateToProps {
    report: Report | null,
    reportFetchStatus: FetchStatus | null,
    reportFetchError: AxiosError | null
}

interface DispatchToProps {
    fetchReport: (reportId: number) => void;
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
    reportId: number;
};

const WorkloadMigrationSummary = asyncComponent(() =>
    import(/* webpackChunkName: "WorkloadMigrationSummary" */ './WorkloadMigrationSummary'));
const InitialSavingsEstimation = asyncComponent(() =>
    import(/* webpackChunkName: "InitialSavingsEstimation" */ './InitialSavingsEstimation'));
const WorkloadInventory = asyncComponent(() =>
    import(/* webpackChunkName: "WorkloadInventory" */ './WorkloadInventory'));

class ReportView extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {
            reportId: props.match.params.reportId
        };
    }

    componentDidMount() {
        const { reportId } = this.state;
        this.props.fetchReport(reportId);
    }

    render() {
        const { report, reportFetchStatus, reportFetchError } = this.props;

        return (
            <ReportViewPage
                report={ report }
                reportFetchStatus={ reportFetchStatus }
                reportFetchError={ reportFetchError }
            >
                <Switch>
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.workloadMigrationSummary}` }
                        component={ WorkloadMigrationSummary }
                    />
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.initialSavingsEstimation}` }
                        render={ (props: any) => <InitialSavingsEstimation { ...props } report={ report } /> }
                    />
                    <Route
                        path={ `${this.props.match.url}/${ReportViewPaths.workloadInventory}` }
                        component={ WorkloadInventory }
                    />

                    <Redirect
                        from={ `${this.props.match.url}` }
                        to={ `${this.props.match.url}/${ReportViewPaths.initialSavingsEstimation}` }
                    />
                </Switch>
            </ReportViewPage>
        );
    }
}

export default ReportView;
