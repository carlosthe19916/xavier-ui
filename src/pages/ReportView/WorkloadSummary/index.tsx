import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import WorkloadMigrationSummary from './WorkloadSummary';
import { GlobalState } from 'src/models/state';
import * as reportActions from '../../../actions/ReportActions';

const mapStateToProps = (state: GlobalState)  => {
    const {
        reportWorkloadSummary,
        reportWorkloadSummaryFetchStatus
    } = state.reportState;
    return {
        reportWorkloadSummary,
        reportWorkloadSummaryFetchStatus
    };
};

const mapDispatchToProps = {
    fetchReportWorkloadSummary: reportActions.fetchReportWorkloadSummary
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(WorkloadMigrationSummary)
);
