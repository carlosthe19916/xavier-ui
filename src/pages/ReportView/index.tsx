import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect, Route, Switch, withRouter } from 'react-router-dom';
import { fetchReport } from '../../actions/ReportActions';
import { RouterGlobalProps } from '../../models/router';
import { Report } from '../../models';
import { GlobalState } from '../../models/state';
import { ReportViewPage } from '../../PresentationalComponents/ReportViewPage';
import WorkloadMigrationSummary from './WorkloadMigrationSummary';
import InitialSavingsEstimation from './InitialSavingsEstimation';
import WorkloadInventory from './WorkloadInventory';

interface StateToProps {
    error: string | null;
    report: Report | null;
    loading: boolean;
}

interface DispatchToProps {
    fetchReport: (reportId: number) => void;
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
    reportId: number;
};

export class ReportView extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {
            reportId: props.match.params.reportId
        };
    }

    componentDidMount(): void {
        const id: number = this.state.reportId;
        if (id) {
            this.props.fetchReport(id);
        }
    }

    render() {
        const { error } = this.props;
        const { reportId } = this.state;

        if (!reportId || error) {
            return <Redirect to={ `/reports` } />;
        }

        return (
            <ReportViewPage reportId={ reportId } history={ this.props.history } match={ this.props.match }>
                <Switch>
                    <Route path={ `${this.props.match.url}/workloadMigrationSummary` } component={ WorkloadMigrationSummary } />
                    <Route path={ `${this.props.match.url}/initialSavingsEstimation` } component={ InitialSavingsEstimation } />
                    <Route path={ `${this.props.match.url}/workloadInventory` } component={ WorkloadInventory } />
                    <Redirect from={ `${this.props.match.url}` } to={ `${this.props.match.url}/workloadMigrationSummary` } />
                </Switch>
            </ReportViewPage>
        );
    }
}

const mapStateToProps = (state: GlobalState)  => {
    let { report, loading, error } = state.reportState;
    return {
        report,
        loading,
        error
    };
};

const mapDispatchToProps = (dispatch: any) => {
    return bindActionCreators({
        fetchReport
    }, dispatch);
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ReportView));
