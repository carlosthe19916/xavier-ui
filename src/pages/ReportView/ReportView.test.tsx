import { shallow } from "enzyme";
import ReportView from "./ReportView";

const props = {
  error: '',
  loading: false,
  report: {
    id: 36,
    customerId: "123456",
    fileName: "file1.json",
    numberOfHosts: 254,
    totalDiskSpace: 5871365,
    totalPrice: 1200,
    creationDate: 254112
  },

  match: {
    params: {
      reportId: 1
    }
  },
  fetchReport: jest.fn(),
  history: null,
  location: null
};

describe("ReportView", () => {
  it("expect to redirect because no reportId", () => {
    const emptyProps = Object.assign({}, props, {
      match: {
        params: {
          reportId: null
        }
      }
    });
    const wrapper = shallow(<ReportView {...emptyProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to redirect because error", () => {
    const errorProps = Object.assign({}, props, {
      error: 'Error loading data'
    });
    const wrapper = shallow(<ReportView {...errorProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render loading object", () => {
    const loadingProps = Object.assign({}, props, {
      loading: true
    });
    const wrapper = shallow(<ReportView {...loadingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render object", () => {
    const objProps = Object.assign({}, props);
    const wrapper = shallow(<ReportView {...objProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
