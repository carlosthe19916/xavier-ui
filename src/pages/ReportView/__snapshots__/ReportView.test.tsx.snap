// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportView expect to redirect because error 1`] = `
<withRouter(ReportViewPage)
  report={
    Object {
      "id": 36,
      "inserted": 254112,
      "lastUpdate": 254112,
      "payloadName": "file1.json",
      "reportDescription": "My report description",
      "reportName": "My report name",
      "status": "CREATED",
    }
  }
  reportFetchStatus={
    Object {
      "error": null,
      "status": "none",
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      key="0"
      path="/reports/1/initialSavingsEstimation"
    />
    <Route
      component={[Function]}
      key="1"
      path="/reports/1/workloadMigrationSummary"
    />
    <Route
      component={[Function]}
      key="2"
      path="/reports/1/workloadInventory"
    />
    <Redirect
      from="/reports/1"
      push={false}
      to="/reports/1/initialSavingsEstimation"
    />
  </Switch>
</withRouter(ReportViewPage)>
`;

exports[`ReportView expect to redirect because no reportId 1`] = `
<withRouter(ReportViewPage)
  report={
    Object {
      "id": 36,
      "inserted": 254112,
      "lastUpdate": 254112,
      "payloadName": "file1.json",
      "reportDescription": "My report description",
      "reportName": "My report name",
      "status": "CREATED",
    }
  }
  reportFetchStatus={
    Object {
      "error": null,
      "status": "none",
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      key="0"
      path="undefined/initialSavingsEstimation"
    />
    <Route
      component={[Function]}
      key="1"
      path="undefined/workloadMigrationSummary"
    />
    <Route
      component={[Function]}
      key="2"
      path="undefined/workloadInventory"
    />
    <Redirect
      from="undefined"
      push={false}
      to="undefined/initialSavingsEstimation"
    />
  </Switch>
</withRouter(ReportViewPage)>
`;

exports[`ReportView expect to render loading object 1`] = `
<withRouter(ReportViewPage)
  report={
    Object {
      "id": 36,
      "inserted": 254112,
      "lastUpdate": 254112,
      "payloadName": "file1.json",
      "reportDescription": "My report description",
      "reportName": "My report name",
      "status": "CREATED",
    }
  }
  reportFetchStatus={
    Object {
      "error": null,
      "status": "none",
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      key="0"
      path="/reports/1/initialSavingsEstimation"
    />
    <Route
      component={[Function]}
      key="1"
      path="/reports/1/workloadMigrationSummary"
    />
    <Route
      component={[Function]}
      key="2"
      path="/reports/1/workloadInventory"
    />
    <Redirect
      from="/reports/1"
      push={false}
      to="/reports/1/initialSavingsEstimation"
    />
  </Switch>
</withRouter(ReportViewPage)>
`;

exports[`ReportView expect to render object 1`] = `
<withRouter(ReportViewPage)
  report={
    Object {
      "id": 36,
      "inserted": 254112,
      "lastUpdate": 254112,
      "payloadName": "file1.json",
      "reportDescription": "My report description",
      "reportName": "My report name",
      "status": "CREATED",
    }
  }
  reportFetchStatus={
    Object {
      "error": null,
      "status": "none",
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      key="0"
      path="/reports/1/initialSavingsEstimation"
    />
    <Route
      component={[Function]}
      key="1"
      path="/reports/1/workloadMigrationSummary"
    />
    <Route
      component={[Function]}
      key="2"
      path="/reports/1/workloadInventory"
    />
    <Redirect
      from="/reports/1"
      push={false}
      to="/reports/1/initialSavingsEstimation"
    />
  </Switch>
</withRouter(ReportViewPage)>
`;
